src/app/pages/forms-and-validation/phone-number-format/phone-number-format.component.ts

```diff
@@ -0,0 +1,118 @@
+import { Component, OnInit } from '@angular/core';
+import { BreakpointObserver, Breakpoints, BreakpointState } from '@angular/cdk/layout';
+import { FormGroup, Validators, FormBuilder, AbstractControl } from '@angular/forms';
+import { StateService } from '../../../services/state.service';
+import { parsePhoneNumberFromString, AsYouType } from 'libphonenumber-js';
+
+@Component({
+    selector: 'app-phone-number-format',
+    templateUrl: './phone-number-format.component.html',
+    styleUrls: ['./phone-number-format.component.scss'],
+})
+export class PhoneNumberFormatComponent implements OnInit {
+    isSmall: boolean;
+    selectedPhoneNumber: any;
+    validatePhoneNumberForm: FormGroup;
+    validPhoneNumber: number;
+    countries: any[] = [
+        { code: 'US', name: '+1 (US)', placeholder: '### ### ####' },
+        { code: 'CA', name: '+1 (CA)', placeholder: '### ### ####' },
+        { code: 'RU', name: '+7 (RU)', placeholder: '### ### ## ##' },
+        { code: 'EG', name: '+20 (EG)', placeholder: '# #######' },
+        { code: 'IN', name: '+91 (IN)', placeholder: '#### ### ###' },
+    ];
+
+    selectedCountry = this.countries[0].code;
+    countryPlaceholder = this.countries[0].placeholder;
+    constructor(
+        private readonly _drawerService: StateService,
+        private readonly _breakpointObserver: BreakpointObserver,
+        private readonly _formBuilder: FormBuilder
+    ) {
+        this.initForm();
+    }
+
+    ngOnInit(): void {
+        this._breakpointObserver
+            .observe([Breakpoints.Small, Breakpoints.Handset])
+            .subscribe((state: BreakpointState) => {
+                if (state.matches) {
+                    this.isSmall = true;
+                } else {
+                    this.isSmall = false;
+                }
+            });
+    }
+
+    initForm(): void {
+        this.validatePhoneNumberForm = this._formBuilder.group({
+            phone: ['', [Validators.required, this.validatePhoneNumberInput.bind(this)]],
+            selectedCountry: ['US', Validators.required],
+        });
+    }
+
+    validatePhoneNumberInput(phoneNumberInput: AbstractControl): any {
+        const inputValue: string = phoneNumberInput.value.toString();
+        const phoneNumber: any = parsePhoneNumberFromString(inputValue, this.selectedCountry);
```

Comment:
can we avoid using the any type here? This is a prime example of a place where knowing the type is important...I have no idea what the output of the parsePhoneNumberFromString function is otherwise.